addons/l10n_it_edi/models/res_company.py

```diff
@@ -33,6 +33,10 @@ class ResCompany(models.Model):
         store=True, readonly=False, help="Fiscal code of your company")
     l10n_it_tax_system = fields.Selection(selection=TAX_SYSTEM, string="Tax System",
         help="Please select the Tax system to which you are subjected.")
+    l10n_it_edi_mode = fields.Selection(selection=[('demo', 'Demo'), ('test', 'Test (experimental)'), ('prod', 'Official')],
+        string="EDI Mode (Italy)", help="EDI operating mode for Italy")
+    l10n_it_proxy_user_id = fields.One2many(comodel_name='account_edi_proxy_client.user',
+        inverse_name='company_id', string="EDI Proxy User (Italy)", domain="[('edi_mode', '=', l10n_it_edi_mode)]")
```

Comment:
I think I understand the problem that you're trying to solve by introducing these fields: each company can be linked to a Demo user, a Test user and an Official user, and additionally we need to identify the active user (which you do here using the l10n_it_edi_mode). 

Still, it's a bit clunky:
- the O2M field is called `l10n_it_proxy_user_id` instead of `_ids` which will be quite confusing for anyone who reads this code. (I understand that the intention is for it to only point to one user, but I still think it's confusing...)
- we need this additional field `l10n_it_edi_mode` in order to identify the active user, which needs to have the same selection options as the `account_edi_proxy_client.user.edi_mode` field, but is not a related field ... Again, I think this will be confusing for anyone who comes by next.

I was thinking we could do things differently: just have a single M2O field that identifies the active proxy user, like so:

```suggestion
    l10n_it_active_proxy_user_id = fields.Many2one("account_edi_proxy_client.user", string="Active EDI Proxy User (Italy)")
```

Maybe I haven't thought about everything though...